porcupine/plugins/rightclick_menu.py

```diff
@@ -17,45 +17,33 @@ def text_is_selected(tab: tabs.FileTab) -> bool:
         return False
     return True
 
+
 def show_menu(event) -> None:
     rm = tkinter.Menu(get_main_window(), tearoff=0)
 
     for path, func in _menu_items.items():
         rm.add_command(label=path, command=func)
 
-
-    flag = bool
-    try:
-        flag=text_is_selected(get_filetab())
-    except AssertionError:
-        flag=False
-    finally:
-        if not flag:
-            for i in _ftab_indexes:
-                rm.entryconfigure(i, state=tkinter.DISABLED)
+    flag = text_is_selected(get_filetab())
+    if not flag:
+        for i in _ftab_indexes:
+            rm.entryconfigure(i, state=tkinter.DISABLED)
 
     rm.tk_popup(event.x_root, event.y_root)
     rm.bind("<Unmap>", (lambda event: rm.after_idle(rm.destroy)), add=True)
 
-def add_rightclick_option(path: str, func, needs_selected_text: bool = False):
+
+def add_rightclick_option(path: str, func, needs_selected_text: bool = False) -> None:
```

Comment:
Let's make mypy happy. It is currently complaining that there is no type annotation for the `func` argument:
```suggestion
def add_rightclick_option(path: str, func: Callable[[tabs.FileTab], object], needs_selected_text: bool = False) -> None:
```
You will need to also add `from typing import Callable`.

Reply:
I have it like that in the case someone wants to add an option that doesn't require the filetab, e.g. someone adding an option for opening a new file.

Reply:
The right-click menu appears when you right-click a text widget of a file tab, so a file tab must exist whenever you are accessing that menu. This means that we always have a filetab that we can pass to the function. If needed, the function can simply ignore the filetab parameter it is getting.